src/wallet/rpcwallet.cpp

```diff
@@ -3546,22 +3547,23 @@ UniValue rescanblockchain(const JSONRPCRequest& request)
     }
 
     int start_height = 0;
-    uint256 start_block, stop_block;
+    Optional<int> stop_height;
```

Comment:
g++ compiler `-Wmaybe-uninitialized` warning:
```
wallet/rpcwallet.cpp: In function ‘UniValue rescanblockchain(const JSONRPCRequest&)’:
wallet/rpcwallet.cpp:3550:19: warning: ‘*((void*)& stop_height +4)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     Optional<int> stop_height;
                   ^~~~~~~~~~~
```

Could be
```suggestion
#include <optional.h>
    Optional<int> stop_height = MakeOptional(false, int());
```
?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/17954#discussion_r395695797

> g++ compiler `-Wmaybe-uninitialized` warning:

This is a known false positive with no side effects from an old compiler. https://www.boost.org/doc/libs/1_72_0/libs/optional/doc/html/boost_optional/tutorial/gotchas/false_positive_with__wmaybe_uninitialized.html

I don't think making code less readable to silence these is a good tradeoff. But if silencing them is actually important, we should at least have an automated check, like a linter or an old gcc running on travis and failing so there doesn't have to be a manual reporting, update, and review cycle each time a new instance turns up.

Reply:
> I don't think making code less readable to silence these is a good tradeoff.

Let me add some context: https://github.com/bitcoin/bitcoin/pull/14711#pullrequestreview-193702611, #15292, #18052

> But if silencing them is actually important, we should at least have an automated check, like a linter or an old gcc running on travis and failing so there doesn't have to be a manual reporting, update, and review cycle each time a new instance turns up.

Do you mean adding of the `-Werror=maybe-uninitialized` option to a compiler on Travis?